system-contracts/contracts/EvmInterpreterLoop.template.yul

```diff
@@ -410,11 +410,10 @@ for { } true { } {
         offset, sp := popStackItemWithoutCheck(sp)
         size, sp := popStackItemWithoutCheck(sp)
 
-        checkMultipleOverflow(offset,size,MEM_OFFSET_INNER(), evmGasLeft)
-        checkMultipleOverflow(destOffset,size,MEM_OFFSET_INNER(), evmGasLeft)
+        checkOverflow(destOffset, size, evmGasLeft)
+        checkMemOverflowByOffset(add(destOffset,size), evmGasLeft)
 
-        // TODO invalid?
-        if or(gt(add(add(offset, size), MEM_OFFSET_INNER()), MAX_POSSIBLE_MEM()), gt(add(add(destOffset, size), MEM_OFFSET_INNER()), MAX_POSSIBLE_MEM())) {
+        if gt(add(add(offset, size), MEM_OFFSET_INNER()), MAX_MEMORY_FRAME()) {
```

Comment:
Are we sure that:
1. offset + size can't overflow?
2. offset + size + MEM_OFFSET_INNER can't overflow?

Reply:
Done!

Reply:
`checkMultipleOverflow` is too much imo, it does more check than needed

Reply:
I'm trying to understand what do we want to achieve by this check. `offset` is an offset in calldata, not memory. I think we may be mixing something here

Reply:
That's right, we mixed calldata and memory, I removed the check.
Do you know if `zksolc` compiles  `calldatacopy` to something that satisfies the `For out of bound bytes, 0s will be copied`. condition specified [here](https://www.evm.codes/#37?fork=cancun)? All semantic tests pass without the check but it'd be good to know if it's because the EraVM has the same behaviour.
Otherwise  we might need to do a check using `calldatasize` to know if the offset goes past to do the memset zero.